### PR TITLE
[CLIENT-8061] Catch `gUM` errors in FF.

### PR DESCRIPTION
**Pull request by mhuynh5757**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### JIRA link(s):

- [CLIENT-8061](https://issues.corp.twilio.com/browse/CLIENT-8061)

### Description

This PR introduces error handling for FF specific `gUM` errors.

### Before review
* [x] Updated CHANGELOG.md if necessary
* [x] Added unit tests if necessary
* [x] Updated affected documentation
* [x] Verified locally with `npm test`
* [x] Manually sanity tested running locally
* [x] Included screenshot as PR comment (if needed)
* [x] Ready for review

### Before merge
* [ ] Got one or more +1s
* [ ] Re-tested if necessary

### CODE REVIEW

**Review by charliesantos — APPROVED** [review]
+1 for me. After getting +1 from Ryan, please don't merge yet because we have a GA release soon. I don't want to block that.

File: lib/AudioInputTest.ts
Comment (ryan-rowland) [inline lib/AudioInputTest.ts:519]: What's with `Error`? I think we can just say:
> An error has occurred.

**Review by ryan-rowland — CHANGES_REQUESTED** [review]

**Review by ryan-rowland — APPROVED** [review]

**Pull request merged**
